_posts/2018-03-08-roadmap.md

```diff
@@ -0,0 +1,221 @@
+    via “non-lexical lifetimes”, improved pattern matching integration, and
+    more.
+-   **Trait system improvements**, including the long-awaited impl Trait syntax
+    for dealing with types abstractly.
+-   **Module system improvements**, focused on increasingly clarity and reducing
+    complexity.
+-   **Generators/async/await:** work is rapidly progressing on first-class async
+    programming support.
+
+In addition, we anticipate a few more major features to stabilize prior to the
+epoch release, including **SIMD**, **custom allocators**, and **macros 2.0**.
+
+### Compiler improvements
+
+As of [Rust 1.24](https://blog.rust-lang.org/2018/02/15/Rust-1.24.html),
+incremental recompilation is available and enabled by default on the stable
+compiler. This feature already makes rebuilds significantly faster than fresh
+builds, but over the course of the year we expect continued improvements for
+_both_ fresh and re-builds. Compiler performance should not be an obstacle to
+productivity in Rust 2018.
+
+
+
+### Tooling improvements
+
+Rust 2018 will see high quality 1.0 releases of the RLS (underlying much of our
+IDE integration story) and rustfmt (a standard formatting tool for Rust
+code). We will continue to improve Cargo by stabilizing custom registries,
+public dependencies, and a revised profile system. We’re also expecting further
+work on [Cargo build system integration], [Xargo integration], and [custom test
+frameworks], though it’s unclear as yet which of these will be complete prior to
+the epoch release.
+
+[Cargo build system integration]: https://github.com/rust-lang/rfcs/pull/2136
+[Xargo integration]: https://github.com/japaric/xargo
+[custom test frameworks]: https://github.com/rust-lang/rfcs/pull/2318
+
+## Library improvements
+
+Building on [our work from last year][blitz], we will publish a 1.0 version of
+the [Rust API guidelines book], continue pushing important libraries to 1.0
+status, improve discoverability through a revamped cookbook effort, and make
+heavy investments in libraries in specific domains—as we’ll see below.
+
+[blitz]: https://blog.rust-lang.org/2017/05/05/libz-blitz.html
+[Rust API guidelines book]: https://github.com/rust-lang-nursery/api-guidelines
+
+**Web site improvements**
+
+As part of Rust 2018, we will completely overhaul the Rust web site, making it
+useful for CTOs and engineers alike. It should be far easier to find information
+to help evaluate Rust for your use case, and to stay up to date with the latest
+tooling and ecosystem improvements.
+
+## Four target domains
+
+Part of our goal with Rust 2018 is to demonstrate Rust’s productivity in
+specific domains of use. We’ve selected four such domains to invest in and
+highlight this year:
+
+- **Network services**. Rust’s reliability and low footprint make it an
+  excellent match for network services and infrastructure, especially at high
+  scale.
+- **CLI apps**. Rust’s portability, reliability, ergonomics, and ability to
```

Comment:
Let's expand CLI before we use the acronym, as not everyone will be familiar with it